### ISSUE TITLE
Time Machine requests slow: they take 7 seconds per request

### ISSUE THREAD

**Issue by levelsio** [Pirate-Weather/pirateweather#115]
### Describe the issue

Not sure if this was always the case but recently time machine requests take 7 seconds.

https://timemachine.pirateweather.net/forecast/REDACTED/13.7566968,100.5019493,1643241600

I need to get historical data for my site Nomad List for ~1500 cities for 365 days, so that'd take 38 days just to get. I think it took less than 1 second before.

The reason I need to get 365 days data is I median/average each month's data to show historical weather per month:
https://nomadlist.com/weather/bangkok

If not fixed, maybe can we get a monthly avg endpoint for this type of request? That'd be 12 requests per city instead of 365 then. Thanks!

### Acknowledgements

- [X] I have searched this repository and [Home Assistant Repository](https://github.com/alexander0042/pirate-weather-ha) to see if the issue has already been reported.
- [X] I have read through the [API documentation](https://pirateweather.net/en/latest/API/) before opening this issue.
- [X] I have written an informative title.

**Comment by alexander0042** [comment]
Thanks for posting this issue, and apologizes for my slow reply here! I appreciate people taking the time to fill out issues, and want to get better at replying to them quickly. 

Time machine does things pretty differently than the forecast api endpoint. Since there's so much data to store, it grabs it out of a public S3 bucket, as opposed to my internal EFS endpoint. The data also isn't optimized for fast reads, and it has to do some fancy on the fly conversion from GRIB into something more useful. 

I do feel like it used to be a bit faster, so while it'll always be slower (unless I manage), I'll check if something slowed down for some reason. 

Another option, depending on your process, would be to run your requests in parallel? Since it's hitting a Lambda function, if there's more requests it'll just spin up more instances, so the (slow) performance should stay roughly the same.

**Comment by levelsio** [comment]
Sure I can do parallel! Thanks. Would it be possible to just create a median monthly value in the API?

Most sites don't care about specific historical data on a day, but want to show a monthly grid to give an idea of the temperatures and climate conditions for that month.

For example my site Nomad List here: https://nomadlist.com/bangkok/weather

All I need for that is median monthly values of temperature, humidity, weather type (summary), precipitation levels, cloudiness, UVI index.

![CleanShot 2023-11-12 at 15 57 18@2x](https://github.com/alexander0042/pirateweather/assets/7150848/cc7b0186-62e8-4e85-86a0-0147262f582c)

**Comment by cloneofghosts** [comment]
I see that this issue has gone stale and I've assigned it to @alexander0042 so it won't go stale in the future. @levelsio Are you still looking to see if median values can be added?

**Comment by cloneofghosts** [comment]
Closing in favour of #130
